brion/csvConfig.cpp

```diff
@@ -73,43 +52,43 @@ struct CsvConfig::Impl
         if (file.fail())
             throw std::runtime_error("Could not open file `" + uri + "`");
 
-        std::string contents;
-
-        file.seekg(0, std::ios::end);
-        contents.reserve(file.tellg());
-        file.seekg(0, std::ios::beg);
+        std::vector<std::string> lines;
 
-        contents.assign((std::istreambuf_iterator<char>(file)),
-                        std::istreambuf_iterator<char>());
+        for (std::string line; std::getline(file, line);)
+            lines.push_back(line);
 
-        const auto lines = explode(contents, '\n');
         for (const auto& line : lines)
-            table.push_back(explode(line, ' '));
-
-        { // Verify layout
-            const size_t numColumns = table.front().size();
-            for (const auto& row : table)
-            {
-                if (numColumns != row.size())
-                    throw std::runtime_error(
-                        "Number of columns inconsistent in `" + uri + "`");
-            }
+            table.push_back(explode_whitespace(line));
+
+        // Verify layout
+        const size_t numColumns = table.front().size();
+        for (const auto& row : table)
+        {
+            if (numColumns != row.size())
+                throw std::runtime_error("Number of columns inconsistent in `" +
+                                         uri + "`");
         }
 
-        { // Fill access maps
-            size_t ctr = 0;
-            for (const auto& columnName : table.front())
-            {
-                nameToColumnIndex[columnName] = ctr;
-                ctr++;
-            }
-
-            for (size_t i = 1; i < table.size(); i++)
-            {
-                const auto& nodeTypeIdStr = table[i][0];
-                const size_t nodeTypeId = std::stoi(nodeTypeIdStr);
-                nodeTypeIdToRowIndex[nodeTypeId] = i;
-            }
+        // Fill access maps
+        size_t ctr = 0;
+        for (const auto& columnName : table.front())
+        {
+            nameToColumnIndex[columnName] = ctr;
+            ctr++;
+        }
+
+        const auto nodeTypeIdIt = nameToColumnIndex.find("node_type_id");
+
+        if (nodeTypeIdIt == nameToColumnIndex.end())
+            throw std::runtime_error("Could not find column 'node_type_id'");
+
+        const auto pos = std::distance(nameToColumnIndex.begin(), nodeTypeIdIt);
```

Comment:
This is not only more complicated that needed, but also incorrect. I'm even surprised the tests are passing.

    const auto pos = *nodeTypeIdIt

Reply:
Yeah, a miracle this worked in the tests...